### ISSUE TITLE
[Component] [input] When inputting Chinese with Pinyin, both Pinyin and Chinese are entered after the input is complete

### ISSUE THREAD

**Issue by crucioo** [element-plus/element-plus#11747]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.47`
- Element Plus Version: `2.1.11`
- Browser / OS: `Chromium/50.0.2661.102`
- Build Tool: `Vite`

### Related Component

- `el-input`

### Reproduction Link

[Link]()

### Steps to reproduce

1. input Chinese with Pinyin
2. complete input
3. the content in the input box contains both pinyin and Chinese

### What is Expected?

The content in the input box only contains the spelled Chinese

### What is actually happening?

the content in the input box contains both pinyin and Chinese

### Additional comments

Tested different versions of elementui and found：
Version 2.1.10 is as expected, However, when upgrading to 2.1.11, the above problems will occur

**Comment by btea** [comment]
I tried it, the input is normal. Please provide a reproduction link or a screenshot of the operation.

**Comment by crucioo** [comment]
> I tried it, the input is normal. Please provide a reproduction link or a screenshot of the operation.

https://user-images.githubusercontent.com/17693532/221747267-e99d82a5-e55e-4cbe-90e3-d060a0cba2f5.mov.

navigator.userAgent: 
waimai_e_pc_client/4.17.0 (Macintosh; Intel Mac OS X 10_16_0) AppleWebKit/537.36 (@c26c0312e940221c424c2730ef72be2c69ac1b67) Chromium/50.0.2661.102 nw/0.14.7 MeituanWaimai/4.17.0

**Comment by btea** [comment]
It looks like you guys are building desktop apps with `nw.js`, do you have this issue in the browser?

**Comment by crucioo** [comment]
> It looks like you guys are building desktop apps with `nw.js`, do you have this issue in the browser?

I just downloaded a lower version of chrome(51.0.2704.84) and tried it, and the same problem happened. It seems that it is not caused by nw.js

https://user-images.githubusercontent.com/17693532/221793058-0bc47c32-e60c-4936-8165-ac7c5838e2ee.mov

**Comment by btea** [comment]
Does the latest version of element-plus still have this problem?

**Comment by crucioo** [comment]
> Does the latest version of element-plus still have this problem?

Yes, version 2.2.32 still has this problem in Chromium/50.0.2661.102

**Comment by btea** [comment]
Does this problem still exist if you update the chrome version?

**Comment by crucioo** [comment]
> Does this problem still exist if you update the chrome version?

升级chrome版本后是没问题的，但是现在我的情况是无法主动升级chrome版本，我用的是别人的壳子(nw.js)，这有什么好的解决办法吗😂

**Comment by btea** [comment]
nw.js 能升级版本吗？不能的话，可以尝试自定义格式化输入内容试试 https://element-plus.org/zh-CN/component/input.html#%E6%A0%BC%E5%BC%8F%E5%8C%96

**Comment by crucioo** [comment]
> nw.js 能升级版本吗？不能的话，可以尝试自定义格式化输入内容试试 https://element-plus.org/zh-CN/component/input.html#%E6%A0%BC%E5%BC%8F%E5%8C%96

验证后发现：[compositionend](https://developer.mozilla.org/zh-CN/docs/Web/API/Element/compositionend_event)在低版本chrome(Chromium/50.0.2661.102)和高版本chrome, 参数中的event.target.value的返回值不一样。低版本返回的是个拼音，高版本返回的是拼音确认后的中文

**Comment by btea** [comment]
这样的话，似乎还是只能升级chrome了

**Comment by crucioo** [comment]
> 这样的话，似乎还是只能升级chrome了

调试了一下，发现这样可以解决上述问题：
```javascript
let tempInput = '';
const compositionstart = (ev) => {
    tempInput = ev.target.value;
};
const compositionend = (ev) => {
    nextTick(() => {
        ElInputRef.value.clear();
        formData.name = tempInput;
    });
};
```
但是我其实没太理解其中的原理，大佬能讲解下吗？

**Comment by winteroo** [comment]
此方法可以保证在低版本浏览器下自动过滤掉拼音

```js
  let initialInputValue = '';

  // 记录 compositionstart 事件时的输入框初始值
  function handleCompositionStart(event: CompositionEvent) {
    initialInputValue = event.target?.value;
  }

  // 处理 compositionend 事件
  function handleCompositionEnd(event: CompositionEvent) {
    // 延时处理，确保输入框值已更新
    setTimeout(() => {
      const finalInputValue = event.target?.value;
      const chineseCharacter = event.data; // 获取输入的汉字

      // 构建拼音部分的正则表达式
      const regex = new RegExp(initialInputValue + '(.*)' + chineseCharacter);
      // 使用正则表达式替换拼音部分为空字符串，只保留汉字
      const filteredValue = finalInputValue.replace(regex, initialInputValue + chineseCharacter);

      // 更新输入框的值
      event.target.value = filteredValue;

      // 手动触发 input 事件以更新组件状态
      const inputEvent = new Event('input', { bubbles: true });
      event.target.dispatchEvent(inputEvent);
    });
  }

  document.addEventListener('compositionend', event => {
    if (event.target?.tagName === 'INPUT') {
      handleCompositionEnd(event);
    }
  });

  document.addEventListener('compositionstart', event => {
    if (event.target?.tagName === 'INPUT') {
      handleCompositionStart(event);
    }
  });
```
